src/components/DateInput/DateInput.tsx

```diff
@@ -92,6 +92,12 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
   const { isOpen, open, close } = useDisclosure();
   const previousDate = usePrevious(dateToDayjs(value, timezone));
 
+  // Handles value & timezone updates outside of the component (i.e. a form has re-initialized or
+  // has updated its values as a result of an API call)
+  React.useEffect(() => {
+    setCurrentDate(dateToDayjs(value, timezone));
```

Comment:
this and the effect below will fire unnecessarily sometimes, but I think I'm unfortunately ok with that. Comparing dayjs/date/undefined to make sure it only fires when things are different seems more bug-prone to me